core/iwasm/interpreter/wasm_interp_classic.c

```diff
@@ -1174,6 +1176,8 @@ wasm_interp_call_func_bytecode(WASMModuleInstance *module,
     uint8 local_type, *global_addr;
     uint32 cache_index, type_index, param_cell_num, cell_num;
     uint8 value_type;
+    bool disable_bounds_checks = !wasm_runtime_is_bounds_checks_enabled(
+        (WASMModuleInstanceCommon *)module);
```

Comment:
Not sure why not use WASM_CONFIGURABLE_BOUNDS_CHECKS to control the code? Now no matter WASM_CONFIGURABLE_BOUNDS_CHECKS is 1 or 0, `disable_bounds_checks` will be read from a function, and compiler cannot optimize the code in CHECK_MEMORY_OVERFLOW/CHECK_BULK_MEMORY_OVERFLOW, it impacts the performance.

Reply:
If WASM_CONFIGUABLE_BOUNDS_CHECKS is disabled, wasm_runtime_is_bounds_checks_enabled  will just return true, so the compiler can do the optmizations.

Reply:
Not very sure, the function isn't inline, my understanding is that some compilers may not be able to optimize it.

How about using WASM_CONFIGUABLE_BOUNDS_CHECKS to control the whole wasm_runtime_is_bounds_checks_enabled:
```C
#if WASM_CONFIGUABLE_BOUNDS_CHECKS  != 0
bool
wasm_runtime_is_bounds_checks_enabled()
{
   ...
}
void
wasm_runtime_set_bounds_checks()
{
    ...
}
```
And in interpreter and wasm_memory.c:

```C
static inline bool
is_bounds_checks_enabled()
{
#if WASM_CONFIGUABLE_BOUNDS_CHECKS  != 0
    return wasm_runtime_is_bounds_checks_enabled();
#else
    return true;
#endif
}
```
And call is_bounds_checks_enabled instead of wasm_runtime_is_bounds_checks_enabled.

Reply:
Or, how about make them to be inline function directly ?

Reply:
Make `is_bounds_checks_enabled` static inline and `wasm_runtime_is_bounds_checks_enabled` non static/inline since the latter may be called as library API?

Reply:
OK